src/Sentry/TransactionTracer.cs

```diff
@@ -313,9 +313,16 @@ private void AddChildSpan(SpanTracer span)
         if (!isOutOfLimit)
         {
             _spans.Add(span);
+            _sortedSpanCache = null; // invalidate cache
```

Comment:
I believe without a memory barrier this won't be visible to all threads at once. the `Interlocked` API can solve that.

Or mark the field as `volatile`

Reply:
Cool, that answers the question I raised in Discord. Thanks Bruno!